core/src/main/scala/zio/logging/CapturedCause.scala

```diff
@@ -0,0 +1,28 @@
+package zio.logging
+
+import zio.Cause
+
+sealed trait CapturedCause {
+  val cause: Cause[Any]
+  def toThrowable: Throwable
+}
+
+object CapturedCause {
+  def apply[E: CauseToThrowable](c: Cause[E]): CapturedCause =
+    new CapturedCause {
+      override val cause: Cause[Any]      = c
+      override def toThrowable: Throwable =
+        implicitly[CauseToThrowable[E]].toThrowable(c)
+    }
+
+  trait CauseToThrowable[-E] {
+    def toThrowable(cause: Cause[E]): Throwable
+  }
+
+  implicit def throwableCause[E <: Throwable]: CauseToThrowable[E] = (cause: Cause[E]) => cause.squash
+  implicit val nothingCause: CauseToThrowable[Nothing]             = (_: Cause[Nothing]) => new RuntimeException("Nothing")
+
+  object causeToRuntimeException {
+    implicit def toThrowable[E]: CauseToThrowable[E] = (cause: Cause[E]) => new RuntimeException(cause.prettyPrint)
```

Comment:
I would probably add this to `Format` handle this. I don't like to make this decision on compile time. Imagine situation that you create library and you decided in your code to use `causeToRuntimeException` and other user try to use your library be it has different idea. If that will be decision that you could make in format it's more flexible.

Reply:
This is a good point, I was not really thinking over how this would be used when logging from a library. Yeah if the library has a custom error type and its author used that fallback instead of providing a proper implementation, then there is no way to override it from the application. On the other hand if it is handled in the formatter, that would require to handle all the possible custom error types in a common log formatter defined in the application, even if it those are implementation detail of the given library that logs it. Maybe this would not be a very common situation, but I still feel that my proposed solution may work better.

Reply:
it will required the same amount of work to do this in format so please add annotation that will carry error and handle this in format. @jdegoes if you have any idea how to make nice api for that please weigh in.

Reply:
I would like to summarize one more time why I believe that my proposed solution to do this on the logger side is better.

I won't talk about _why_ is it good to be able to have a conversion to Throwable because I think we are on the same page about that. If not I'm happy to write in more details about that one too.

Let's imagine a situation of writing a service that uses a (imagined version of) `zio-web` to implement a HTTP API with some library-provided, zio-logging based logging middleware. It also uses `zio-aws` to work with some AWS services such as S3 and also have its own debug logging for that. It can also use an internal company specific library to work with some database let's say, with company standard logging included, and the service itself has it's own business logic. In this example I would imagine the HTTP library has an error type that is somehow the union of some HTTP errors and the middleware errors, the AWS library has it's own error type, let's say the SQL library works with Throwables as it uses JDBC underneath, and the service business logic also has it's own typed errors. 

In this example each component would use zio-logging independently to log errors of different types. 

With my proposed solution each library's developer can decide to use the deault conversion or provide a more specific one. For example one of the error cases wraps exceptions coming from the AWS Java library, so that can be converted in a better way to throwable, keeping the nested exception. This is something that should be decided by the developer of the library, and written once. With that version the service developer just pulls in the various libraries using logging, provide a logging layer and everything just works automatically.

On the other hand, if I understand correctly, you propose to handle this conversion in the log formatter. I see multiple problems with that.

First of all, as I said, the error type gets erased currently. I see three ways to fix that. Either capture the conversion itself at the entry point (this is my proposal), or just capture a type tag, or convert the annotations to a HList or similar. 

Let's try to imagine what it would look like if we choose one of the other possibilities.

If we just store a `Cause[_]` and a type tag representing `E`, in your formatter you have to check the type tag to "reverse engineer" the original base error type, cast the cause, and then do the desired conversion. And every time you add anything new to the service, include a new library, or someone writes a new functionality, this has to be reviewed because there is no way to compile time check that all the possible `E`s are covered. And you need to replicate the knowledge of what a 3rd party library is logging in each application that uses that library.

With a `HList`-like structure you could have a more or less convenient way to choose is to select an implicit conversion matching the `E` of the annotation's `Cause` on the formatter side. But to do that, you would have to parametrize on the HList type all the way down to the formatter, which is much more invasive than capturing at the log entrypoint. And even if it would be done, it has the same problem I mentioned earlier of having a central place to know about all the error types in the application. (Although with compile time type checks, with most likely very hard to understand errors for incomplete definitions)

So I really don't see how that could work. If you have any idea that I missed, please let me know.